src/Extensions/Error.test.ts

```diff
@@ -0,0 +1,46 @@
+import { describe } from 'riteway'
+import './Error'
+
+describe('Extensions Error', async (should: any) => {
+  const { assert } = should('')
+
+  {
+    const customError = new Error('error')
+    const actual = JSON.parse(JSON.stringify(customError))
+
+    assert({
+      given: 'a new Error',
+      should: 'transform to an valid object',
+      actual: typeof actual === 'object',
+      expected: true
+    })
+
+    assert({
+      given: 'a new Error',
+      should: 'get the keys of the new Error',
+      actual: Object.keys(actual),
+      expected: ['stack', 'message', 'type']
+    })
+
+    assert({
```

Comment:
There's a few more cases we may want to add if we want to be super thorough: the `type` property is set to the Error's constructor name if there isn't a set `type` property already.

So, for example, `new Error` will have `.type === 'Error'`, `new MyCustomError` will have `.type === 'MyCustomError` and if we do something like...

```ts
class MyCustomError extends Error {
  constructor(...) {
    this.type = 'SomeType'
  }
}
const err = new MyCustomError()
console.log(JSON.stringify(err)) // { ..., "type": "SomeType" }
```

This was done like this because MongoDB errors come with a `type` field while other errors don't.

And this only happens in the `toJSON` because native JS Error doesn't come with a decent `toJSON` method, and PinoJS (and other loggers) use the `.toJSON` method of passed objects to convert them to strings before sending them to the logging output.

Covering these cases will probably increase coverage, since the conditional in the toJSON may be considered as different branches of code.

It could also be nice to make explicit the fact that we're testing the `toJSON` method (in the cases we are). Unit tests also serve as good documentation (by declaring intent).

We can do this in a separate PR or open an issue if you prefer, either way works fine.

Reply:
Extending base objects in javascript is an anti-pattern and best avoided.
I would prefer to see us get the bare minimum of unit tests in place that will support a future refactoring away from these extensions.

Reply:
@warrenv I've been meaning to add documentation on why and how we're extending base JS objects. It's completely safe if done right.

We still may change a few of this in favor of composition with Ramda, but `Error.toJSON` in particular may live with us for a while. 

On the other hand, even if we stop binding these functions to base objects, we'll still have them, so we'll still want the unit tests.

Reply:
@lautarodragan added it!